DashboardFrontend/Controller.cs

```diff
@@ -354,56 +350,141 @@ private void GetCredentials()
+        /// </summary>
+        /// <remarks><see cref="Thread.Sleep(int)"/> is used to drastically reduce the number of times the loop is executed, lowering the overall CPU load.</remarks>
+        private void UpdateViews()
+        {
+            while (_vm.IsRunning)
             {
-                foreach(Execution newExec in result)
+                if (Conversion?.ActiveExecution is null)
                 {
-                    if (!Conversion.Executions.Any(e => e.Id == newExec.Id))
+                    continue;
+                }
+                foreach (ValidationReportViewModel vm in ValidationReportViewModels)
+                {
+                    if (vm.LastUpdated < Conversion.LastValidationsUpdated)
                     {
-                        Conversion.AddExecution(newExec);
+                        vm.LastUpdated = DateTime.Now;
+                        Application.Current.Dispatcher.Invoke(() =>
+                        {
+                            vm.UpdateData(Conversion.ActiveExecution.ValidationReport);
+                        });
+                    }
+                }
+                foreach (LogViewModel vm in LogViewModels)
+                {
+                    if (vm.LastUpdated < Conversion.LastLogUpdated)
+                    {
+                        vm.LastUpdated = DateTime.Now;
+                        Application.Current.Dispatcher.Invoke(() =>
+                        {
+                            vm.UpdateData(Conversion.ActiveExecution.Log);
+                        });
+                    }
+                }
+                foreach (ManagerViewModel vm in ManagerViewModels)
+                {
+                    if (Conversion?.ActiveExecution is not null && vm.LastUpdated < Conversion?.LastManagerUpdated)
+                    {
+                        vm.LastUpdated = DateTime.Now;
+                        Application.Current.Dispatcher.Invoke(() =>
+                        {
+                            vm.UpdateData(Conversion.ActiveExecution.Managers);
+                        });
                     }
                 }
+                foreach (HealthReportViewModel vm in HealthReportViewModels)
+                {
+                    if (vm.LastUpdated < Conversion.LastHealthReportUpdated)
+                    {
+                        vm.LastUpdated = DateTime.Now;
+                        Application.Current.Dispatcher.Invoke(() =>
+                        {
+                            vm.SystemLoadChart.UpdateData(Conversion.HealthReport.Ram, Conversion.HealthReport.Cpu);
+                            vm.NetworkChart.UpdateData(Conversion.HealthReport.Network);
+                            vm.NetworkDeltaChart.UpdateData(Conversion.HealthReport.Network);
+                            vm.NetworkSpeedChart.UpdateData(Conversion.HealthReport.Network);
+                        });
+                    }
+                }
+                Thread.Sleep(10);

```

Comment:
kan den her ikke skiftes til 100? 
```suggestion
                Thread.Sleep(100);

```